sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java

```diff
@@ -139,23 +141,49 @@
          * Expected UserGroups that an authority will be granted to if found in the response from the MemeberOf Graph
          * API Call.
          */
-        private List<String> allowedGroups = new ArrayList<>();
+        private List<String> allowedGroupNames = new ArrayList<>();
 
+        private Set<String> allowedGroupIds = new HashSet<>();
+
+        public Set<String> getAllowedGroupIds() {
+            return allowedGroupIds;
+        }
+
+        public void setAllowedGroupIds(Set<String> allowedGroupIds) {
+            this.allowedGroupIds = allowedGroupIds;
+        }
+
+        public List<String> getAllowedGroupNames() {
+            return allowedGroupNames;
+        }
+
+        public void setAllowedGroupNames(List<String> allowedGroupNames) {
+            this.allowedGroupNames = allowedGroupNames;
+        }
+
+        @Deprecated
```

Comment:
test @Deprecated(since="xxx")

Reply:
@backwind1233  @ZhuXiaoBing-cn should we use `@DeprecatedConfigurationProperty` here?